pkg/manifestcomparators/comp_no_field_removal.go

```diff
@@ -22,6 +22,26 @@ func (noFieldRemoval) WhyItMatters() string {
 	return "If fields are removed, then clients that rely on those fields will not be able to read them or write them."
 }
 
+func getFieldsAndEnums(version *apiextensionsv1.CustomResourceDefinitionVersion) (sets.String, map[string]sets.String) {
+	fields := sets.NewString()
+	enumsMap := make(map[string]sets.String)
+	SchemaHas(version.Schema.OpenAPIV3Schema, field.NewPath("^"), field.NewPath("^"), nil,
+		func(s *apiextensionsv1.JSONSchemaProps, fldPath, simpleLocation *field.Path, _ []*apiextensionsv1.JSONSchemaProps) bool {
+			fields.Insert(simpleLocation.String())
+			for _, enum := range s.Enum {
```

Comment:
This is a validation rule, not a new location right?  I'm in favor of the rule, I'd prefer a new file to check for "no_enumeration_removal" versus checking both fields and validation from one checker

Reply:
I moved the check to a new file.